pkg/api/v1/clients/kube/resource_client.go

```diff
@@ -457,8 +459,28 @@ func (rc *ResourceClient) convertCrdToResource(resourceCrd *v1.Resource) (resour
 				*status = typedStatus
 				return nil
 			}
-			if err := resources.UpdateStatus(withStatus, updateFunc); err != nil {
-				return nil, err
+			updateReporterStatusFunc := func(status *core.ReporterStatus) error {
+				if status == nil {
+					return nil
+				}
+				typedStatus := core.ReporterStatus{}
+				if err := protoutils.UnmarshalMapToProto(resourceCrd.Status, &typedStatus); err != nil {
+					return err
+				}
+				*status = typedStatus
+				return nil
+			}
+			// First attempt to unmarshal Status
+			if statusErr := resources.UpdateStatus(withStatus, updateStatusFunc); statusErr != nil {
+				// If unmarshalling Status failed, the resource likely has a ReporterStatus instead.
```

Comment:
let's switch the order here? going forward, it will be more likely we see a reporter status, so no reason to try and fail to unmarshal it as a status all the time. slightly more efficient on the cpu in more cases

Reply:
Good idea, thanks